proposals/5/README.md

```diff
@@ -0,0 +1,103 @@
+****---
+title: Decomposition of Jenkins X
+linktitle: Decomposition of Jenkins X
+description: Decomposition of Jenkins X
+type: docs
+weight: 50
+---
+
+## 1. Overview
+
+This document outlines the work for decomposing Jenkins X into a series of reusable libraries.
+
+## 1.1 Motivation
+
+At the moment, the source code that comprises the main JX binary is tightly coupled and difficult to work on.  Changes have
+unintended consequences and this makes the codebase fragile.
+
+One of the design goals in Accelerate is to have a **Loosely Coupled Architecture** - in particular: 
+
+_The goal is for your architecture to support the ability of teams to get their work done—from design through to deployment—without requiring high-bandwidth communication between teams._
```

Comment:
If the goal is to reduce communication, is decomposition really the way to go? I would argue that the modularized architecture will increase the need for communication since it will actually be harder to evolve a given component. Also, we are talking about decomposition on a library level opposed decomposing services where you define an API which you agree to adhere to whereas implementation details can change.

I think that having this single point "monolith" makes things easier in terms of communication. jx has architecturally several advantages, the problem is the execution and all the issues which come from using it as a build time dependency.

Each split out component needs to carry its weight since it includes overhead in managing it.

Reply:
Sorry, this was a direct quote from Accelerate, I was using it to try and highlight the value of a loosely coupled architecture.

Reply:
not sure whether it fits in this context, maybe in the bigger context your are alluding to above.